share/trick/trickops/TrickWorkflow.py

```diff
@@ -390,11 +391,22 @@ def type_expected(var, expected_type, fatal=False, extra_msg=''):
             sanitized_labels =( [l for l in c[s]['labels'] if type_expected(
               l, expected_type=str, extra_msg='Ignoring label "%s".' % l) ] )
             self.config[s]['labels'] = sanitized_labels
+          # Check for binary argument
+          if 'binary' not in c[s] or not c[s]['binary']:
+            self.config[s]['binary'] = self.defaults['binary']
+
+          # Add the CPU format string (won't fail if there is no CPU placeholder)
+          self.config[s]['binary'] = self.config[s]['binary'].format(cpu=self.trick_host_cpu)
+
+          # Add the full path to the build command
+          self.config[s]['build_command'] = os.path.join(self.trick_dir, "bin", self.config[s]['build_command'])
+
```

Comment:
I attach the full path and format the binary string here while the configuration is being processed, which is maybe not the best way to do it. The main issue is that if a user creates a run or sim manually in their script, neither of those things will be applied. This is a problem related to the class structure of TrickOps - the Sim and Run class do not have access to the trick directory or host cpu (although that could be obtained if it had the trick_directory). I couldn't figure out a way around that without either using globals or changing the interface which I am hesitant to do. 

Thoughts?

Reply:
Or maybe it would be better to take the build_cmd option away all together and have build_args or something instead? That doesn't solve the data access issue, but it would make the user interface a little cleaner I think.

Reply:
Agree that `build_args` probably makes more sense if we've determined there's no use for `build_cmd` beyond specifying `trick-CP <args>`. I'm all for cleaning up the interface and making the class architecture more useful. TrickOps is very young and this type of critical review from other Trick users / developers if exactly what the code needs.  Right now my teams are the only ones using TrickOps, so we can easily adapt to any framework changes that come out of this review process. All that to say, let's change TrickOps to make it better, don't worry about affecting downstream uesrs, I represent all of them!